### PR TITLE
chore: Remove ruby2.7 support

### PR DESCRIPTION
**Pull request by hnnasit**
Note: Dev canaries were run on this PR. https://ci.appveyor.com/project/AWSSAMCLI/aws-sam-cli-canary-linux-dev/builds/49147181 and https://ci.appveyor.com/project/AWSSAMCLI/aws-sam-cli-canary-windows-dev/builds/49147185
#### Why is this change necessary?

Deprecate runtime ruby2.7 already deprecated by AWS Lambda

#### How does it address the issue?

Removes all occurrences of ruby2.7

#### What side effects does this change have?

No side effects

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [x] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -148,9 +148,6 @@ jobs:
       - uses: actions/setup-go@v5
         with:
           go-version: '1.19'
-      - uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: "2.7"
```
Comment (mndeveci) [inline .github/workflows/build.yml:153]: Anything needed for appveyor configurations?

File: tests/integration/validate/test_validate_command.py [hunk 2]
```diff
@@ -149,7 +149,6 @@ def test_lint_supported_runtimes(self):
             "python3.8",
             "python3.9",
             "python3.10",
-            "ruby2.7",
```
Comment (mndeveci) [inline tests/integration/validate/test_validate_command.py:152]: How about adding `ruby3.2` here?

**Review by mndeveci — COMMENTED** [review]
LGTM in general, couple of small comments

File: .github/workflows/build.yml (hunk 1 above)
Comment (hnnasit) [reply .github/workflows/build.yml:153]: Nope, ruby3.2 is already being setup on lines 151-153. I ran dev canaries and all the tests are passing.

**Review by bentvelj — APPROVED** [review]

**Review by mndeveci — APPROVED** [review]

**Review by hawflau — APPROVED** [review]

**Pull request merged**